python_modules/libraries/dagster-dg/dagster_dg/cli/init.py

```diff
@@ -38,20 +39,44 @@
     type=click.Choice(["persistent_uv", "active"]),
     help="Type of Python environment in which to launch subprocesses for the project.",
 )
+@click.argument(
+    "dirname",
+    required=False,
+)
 @cli_telemetry_wrapper
 def init_command(
+    dirname: Optional[str],
     use_editable_dagster: Optional[str],
-    workspace_name: Optional[str],
+    workspace: bool,
     project_name: Optional[str],
     project_python_environment: DgProjectPythonEnvironment,
     **global_options: object,
 ):
-    """Initialize a new Dagster project or workspace.
+    """Initialize a new Dagster project, optionally inside a workspace.
+
+    By default, only a standalone project is created. The project will be created in a new directory specified by DIRNAME. If DIRNAME is not specified, the user will be prompted for it.
 
-    By default, only a standalone project is created. A workspace can also be created using the
-    --workspace-name option.
+    If the --workspace flag is passed, a workspace will be created in DIRNAME and a project created inside the workspace. If DIRNAME is not passed, the user will be prompted for it. The project name can be specified with the --project-name option, or the user will be prompted for it.
 
-    If a project is created, it will have the following structure:
+    In either case, "." can be passed as DIRNAME to create the new project or workspace inside the existing working directory.
+
+    Examples:
+        dg init
+            Scaffold a new project in a new directory `<project_name>`. Prompts for `<project_name>`.
+        dg init .
+            Scaffold a new project in the CWD.
```

Comment:
assume project name is taken from the name of the last component of the  CWD?

Reply:
yes